connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -392,6 +401,10 @@ protected Consumer<K, V> createConsumer() {
         return new KafkaConsumer<>(consumerConfigs);
     }
 
+    protected boolean readPartition(TopicPartition topicPartition) {
```

Comment:
Nit: add a Javadoc explaining when this method will be used, what it will be used for, and that it's designed to be overridden by subclasses?